ddl/ddl.go

```diff
@@ -464,7 +464,10 @@ func (d *ddl) doDDLJob(ctx context.Context, job *model.Job) error {
 			return nil
 		}
 
-		return errors.Trace(historyJob.Error)
+		if historyJob.Error != nil {
```

Comment:
I think this doesn't a good way to handle this problem.
If historyJob isn't synced, the historyJob's Error mustn't be nil. So I think something wrong with the operation of running DDL job.

Reply:
Just change one line, where I think the error handling could be improved.
I think **We should never errors.Trace(xxx) when xxx is other than error type.**.

Maybe my use the wrong presentation, **this have nothing to do with fix**, I just want to make the code more robust. So is that acceptable? @zimulala

Reply:
No. 
I think historyJob's Error is nil is also a problem.

Reply:
Relate to #5751